mitodl/mit-open
# Cc/no merge generated

#### What are the relevant tickets?
None.

#### What's this PR do?
This PR:
1. Marks generated files as
    - `linguist-generated`, so that github does not show the diffs by default 
    - `merge=binary`, so that generated files cannot be merged. _You should regenerate them instead._
2. Regenerates OpenAPI, since it became out of date due to merges of #34 and #60 

#### Context
The generated OpenAPI code on `main` is out of date. This happened because:
- #34 changed how `LearningResourceViewSet` handles filters, as a consequence, `drf-spectacular` began picking up the filter parameters for **all listing** APIs inhering from it.
- #60 fixed a bug where `/upcming` and `/new` were not being treated as listing APIs.

Both PRs changed `openapi.yaml`. #60 was merged first, and #34 was merged without a rebase. Git did not force us to rebase since the changes to `openapi.yaml` could be merged by git's normal rules.

#### How should this be manually tested?
You could manually test this via:

1. `git checkout cc/no-merge-generated`
2. `git checkout -b test-a `
3. Add a comment to the top of openapi.yaml, and commit the change on branch `test-a` 
4. `git checkout cc/no-merge-generated`
5. Add a comment on the bottom of openapi.yaml and commit the change.
    - Before committing, check `git diff`. You should still get a meaningful diff.
6. `git merge test-a` ... you should get a merge conflict.
```
